ansible/deploy.yml

```diff
@@ -239,15 +239,24 @@
           port: "{{ node_config.http.external.port }}"
           host: "{{ public_ipv4 }}"
           timeout: 300
-        when: health_check
+        when: health_check and not api_gateway
         connection: local
         tags: [health-check]
 
       - name: API health check
         uri:
           url: "{{ api_base_uri }}/blocks/top"
           timeout: 30
-        when: health_check
+        when: health_check and not api_gateway
+        connection: local
+        tags: [health-check]
+
+      - name: Wait aeternity node API to boot gateway
+        wait_for:
+          port: "{{ node_config.http.external.port }}"
+          host: "localhost"
+          timeout: 300
+        when: api_gateway
         connection: local
```

Comment:
Are you sure `connection: local` works with host:localhost ? I think that this task will be run locally on the controller host and do a request to the controller host